### PR TITLE
Index storage bug fix

### PR DESCRIPTION
**Pull request by franzmpub**
# Pull Request      

## What does this PR do?

The change follows the bug fix [here](https://github.com/stanford-futuredata/ColBERT/commit/b3a80aa37b1bac01fbdb0ef421c23e537885c574)

The purpose is to put the `approx_scores` to GPU in GPU-based runs.

Without this, the `torch.topk(approx_scores, k=config.ndocs)` call fails on a GPU.
 
### Request Review

Be sure to **request a review** from one or more reviewers (unless the PR is to an unprotected branch).

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [x] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [ ] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [x] Added [description](#description) of changes proposed.
- [x] Review requested as [appropriate](#request-review).
- [x] Version bumped as [appropriate](#Versioning).
- [ ] New classes, methods, and functions documented.
- [ ] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

**Review by bhavani105 — APPROVED** [review]
LGTM

**Review by jonsaadfalcon — APPROVED** [review]

File: primeqa/ir/dense/colbert_top/colbert/search/index_storage.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def score_pids(self, config, Q, pids, centroid_scores):
                 codes_packed_ = codes_packed[idx_]
                 approx_scores_ = centroid_scores[codes_packed_.long()]
                 if approx_scores_.shape[0] == 0:
-                    approx_scores.append(torch.zeros((len(pids_),), dtype=approx_scores_.dtype))
```
Comment (jsmcibm) [inline primeqa/ir/dense/colbert_top/colbert/search/index_storage.py:137]: shouldn't this be wrapped in `if self.use_gpu:`?

Comment (jsmcibm) [reply primeqa/ir/dense/colbert_top/colbert/search/index_storage.py:137]: there are other examples of this in the same file

Comment (franzmpub) [reply primeqa/ir/dense/colbert_top/colbert/search/index_storage.py:137]: > shouldn't this be wrapped in `if self.use_gpu:`?

Isn't it already (line 123)?

Comment (franzmpub) [reply primeqa/ir/dense/colbert_top/colbert/search/index_storage.py:137]: > there are other examples of this in the same file

Please say more, where in the file?

**Review by jsmcibm — APPROVED** [review]
lgtm

**Pull request merged**